internal/k8s/deployments_test.go

```diff
@@ -557,3 +566,246 @@ func TestComputeConnectionSpecHash(t *testing.T) {
+			container := deployment.Spec.Template.Spec.Containers[0]
+			envMap := make(map[string]string)
+			for _, env := range container.Env {
+				envMap[env.Name] = env.Value
+			}
+
+			for expectedKey, expectedValue := range tt.expectedEnvVars {
+				actualValue, exists := envMap[expectedKey]
+				assert.True(t, exists, "Environment variable %s should be present", expectedKey)
+				assert.Equal(t, expectedValue, actualValue, "Environment variable %s should have correct value", expectedKey)
+			}
+
+			// Verify unexpected environment variables are not present
+			for _, unexpectedKey := range tt.unexpectedEnvVars {
+				value, exists := envMap[unexpectedKey]
+				assert.False(t, exists, "Environment variable %s should not be present, but found value: %s", unexpectedKey, value)
+			}
+
+			// For mTLS case, verify volume mounts and volumes are configured
+			if tt.connection.MutualTLSSecret != "" {
+				assert.Len(t, container.VolumeMounts, 1)
+				assert.Equal(t, "temporal-tls", container.VolumeMounts[0].Name)
+				assert.Equal(t, "/etc/temporal/tls", container.VolumeMounts[0].MountPath)
+
+				assert.Len(t, deployment.Spec.Template.Spec.Volumes, 1)
+				assert.Equal(t, "temporal-tls", deployment.Spec.Template.Spec.Volumes[0].Name)
+				assert.Equal(t, tt.connection.MutualTLSSecret, deployment.Spec.Template.Spec.Volumes[0].VolumeSource.Secret.SecretName)
+			}
+		})
+	}
+}
+
+// createTestCerts generates a self-signed certificate and private key for testing purposes.
+// WARNING: This uses a lighter-weight 1024-bit RSA key for faster test execution.
+// DO NOT copy this for production use - use 2048-bit or higher keys for security.
+func createTestCerts(t *testing.T, tempDir string) (certPath, keyPath string) {
+	t.Helper()
+
+	certPath = filepath.Join(tempDir, "client.pem")
+	keyPath = filepath.Join(tempDir, "client.key")
+
+	// Generate a self-signed certificate for testing (using 1024-bit for speed)
+	privateKey, err := rsa.GenerateKey(rand.Reader, 1024)
+	require.NoError(t, err)
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName: "test",
+		},
+		NotBefore:   time.Now(),
+		NotAfter:    time.Now().Add(time.Hour),
+		KeyUsage:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &privateKey.PublicKey, privateKey)
+	require.NoError(t, err)
+
+	// Write certificate file
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certDER})
+	err = os.WriteFile(certPath, certPEM, 0644)
+	require.NoError(t, err)
+
+	// Write private key file
+	keyDER, err := x509.MarshalPKCS8PrivateKey(privateKey)
+	require.NoError(t, err)
+	keyPEM := pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: keyDER})
+	err = os.WriteFile(keyPath, keyPEM, 0644)
+	require.NoError(t, err)
```

Comment:
Is there an option to write directly to a file/io.Writer instead of the intermediate write to memory?